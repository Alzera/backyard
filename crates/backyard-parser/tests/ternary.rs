use backyard_parser::parse_eval;

#[test]
fn basic() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "$a = false ? 1 : 2;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}
