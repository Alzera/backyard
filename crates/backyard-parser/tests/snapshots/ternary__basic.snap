---
source: crates/backyard-parser/tests/ternary.rs
expression: asts
snapshot_kind: text
---
node_type: program
children:
  - node_type: assignment
    left:
      node_type: variable
      name:
        node_type: identifier
        name: a
        loc:
          start:
            line: 1
            column: 0
            offset: 0
          end:
            line: 1
            column: 1
            offset: 1
        leadings: ~
        trailings: ~
      loc:
        start:
          line: 1
          column: 0
          offset: 0
        end:
          line: 1
          column: 1
          offset: 1
      leadings: ~
      trailings: ~
    operator: "="
    right:
      node_type: ternary
      condition:
        node_type: boolean
        is_true: false
        loc:
          start:
            line: 1
            column: 5
            offset: 5
          end:
            line: 1
            column: 5
            offset: 5
        leadings: ~
        trailings: ~
      valid:
        node_type: number
        value: "1"
        loc:
          start:
            line: 1
            column: 13
            offset: 13
          end:
            line: 1
            column: 13
            offset: 13
        leadings: ~
        trailings: ~
      invalid:
        node_type: number
        value: "2"
        loc:
          start:
            line: 1
            column: 17
            offset: 17
          end:
            line: 1
            column: 17
            offset: 17
        leadings: ~
        trailings: ~
      loc:
        start:
          line: 1
          column: 11
          offset: 11
        end:
          line: 1
          column: 17
          offset: 17
      leadings: ~
      trailings: ~
    loc:
      start:
        line: 1
        column: 3
        offset: 3
      end:
        line: 1
        column: 17
        offset: 17
    leadings: ~
    trailings: ~
loc:
  start:
    line: 1
    column: 0
    offset: 0
  end:
    line: 1
    column: 18
    offset: 18
leadings: ~
trailings: ~
